src/main/java/com/example/crm_gym/repository/TrainingDAOImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.example.crm_gym.repository;
+
+import com.example.crm_gym.dao.TrainingDAO;
+import com.example.crm_gym.models.Training;
+import com.example.crm_gym.storage.Storage;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Repository
+public class TrainingDAOImpl implements TrainingDAO {
+
+    private static final Logger logger = LoggerFactory.getLogger(TrainingDAOImpl.class);
+
+    private final Storage storage;
+    private Map<Integer, Training> trainingStorage;
```

Comment:
What is the reason for having this Map in this class and in the Storage class at the same time???